MobiusCore/Test/EffectHandlers/EffectHandlerDSLTests.swift

```diff
@@ -36,26 +36,89 @@ private enum Event: Equatable {
 class EffectRouterDSLTests: QuickSpec {
     // swiftlint:disable function_body_length
     override func spec() {
+        context("An EffectHandler which always ends as soon as it is called") {
+            var wasDisposed: Bool!
+            var connection: Connection<Int>!
+
+            beforeEach {
+                wasDisposed = false
+                connection = EffectRouter<Int, ()>()
+                    .routeEffects(equalTo: 1).to { _, response in
+                        response.end()
+                        return AnonymousDisposable {
+                            wasDisposed = true
+                        }
+                    }
+                    .asConnectable
+                    .connect { _ in }
+            }
+
+            it("should not be disposed if end was called") {
+                connection.accept(1)
+                connection.dispose()
+
+                expect(wasDisposed).to(beFalse())
+            }
+
+            it("should be disposed if end was never called") {
```

Comment:
The test description doesn't match the implementation. Is it testing the right thing?